test/cat.jl

```diff
@@ -57,11 +49,24 @@ module TestCat
     @testset "hcat ::Vectors" begin
         df = DataFrame()
         DataFrames.hcat!(df, CategoricalVector{Union{Int, Missing}}(1:10))
+        @test df[1] == collect(1:10)
         @test df[1] == CategoricalVector(1:10)
-        DataFrames.hcat!(df, collect(1:10))
+        DataFrames.hcat!(df, 1:10)
```

Comment:
As the name of the set says "::Vectors", you should keep `collect` here and below (or change the name).

Reply:
it was not my code so I did not want to mess with it, fixed to `AbstractVector` test as this is what we really want to test